spec/factories/page.rb

```diff
@@ -1,5 +1,15 @@
 FactoryBot.define do
   factory :page, class: "Form::Page" do
+    id { "page_id" }
     initialize_with { new(id, nil, nil) }
+    trait :with_question do
+      transient do
+        question_id { nil }
+
+        after :build do |page, evaluator|
+          page.questions = [build(:question, id: evaluator.question_id, page:)]
+        end
+      end
+    end
```

Comment:
the point of making all these call downwards recursively is that on line 10 we assign the page to the question (and something similar in the other factories)
which allows code of the form [sic] `question.form` to work.
This isn't actually necessary for the test I am writing here, but I think it's an important pattern to demonstrate for this model of building forms on the fly for tests